cunumeric/deferred.py

```diff
@@ -1966,7 +1993,9 @@ def where(self, src1: Any, src2: Any, src3: Any) -> None:
 
     # A helper method for attaching arguments
     def add_arguments(
-        self, task: Any, args: Optional[Sequence[npt.NDArray[Any]]]
+        self,
+        task: Union[AutoTask, ManualTask],
```

Comment:
You might expect this to be `task: Task`, but `Task` does not define `add_input`.

Reply:
Is it worth inserting a parent class in between to denote tasks that provide this interface?

Reply:
So actually it looks like `AutoTask` and `ManualTask` define their `add_input` method in incompatible ways:

```
    def add_input(
        self, store: Store, partition: Optional[PartSym] = None
    ) -> None:
```

vs

```
    def add_input(
        self,
        arg: Union[Store, StorePartition],
        proj: Optional[ProjFn] = None,
    ) -> None:
```

and the `add_arguments` function takes advantage of the fact that both `add_input` methods do the same when passing in scalar values. Therefore I actually don't think we need to bother with this.